scripts/watchNative.js

```diff
@@ -0,0 +1,201 @@
+      } else {
+        console.log(
+          "\t[tasks -> golang] go build failed ✘\nerror: \n",
+          errText
+        );
+      }
+      resolve();
+    })
+    .on("data", (data) => {})
+    .stderr.on("data", (data) => {
+      errText += data;
+    });
+
+  stream.end(
+    `cd ${p.posix.join(config.deployDirectory, "golang")} && go build\nexit\n`
+  );
+}
+
+async function deleteFile(remotePath) {
+  return new Promise((resolve, reject) => {
+    if (!sshReady) {
+      reject(new Error("SSH connection not ready"));
+      return;
+    }
+
+    conn.sftp((err, sftp) => {
+      sftp
+        .deleteFile(remotePath)
+        .then((res) => resolve())
+        .catch((err) => console.log(err) && resolve());
+    });
+  });
+}
+
+async function uploadFile(localPath, remotePath) {
+  return new Promise((resolve, reject) => {
+    if (!sshReady) {
+      reject(new Error("SSH connection not ready"));
+      return;
+    }
+
+    conn.sftp((err, sftp) => {
+      if (err) {
+        reject(err);
+        return;
+      }
+
+      process.stdout.write(` -> ${remotePath} `);
+      const readStream = fs.createReadStream(
+        p.join("native", localPath),
+        "utf8"
+      );
+      const writeStream = sftp.createWriteStream(remotePath, {
+        encoding: "utf8",
+      });
+
+      writeStream.on("close", () => {
+        conn.shell((err, stream) => {
+          // If the uploaded file is in the `c` directory, run `make`
+          if (localPath.split(p.sep)[0] === "c") {
+            cTask(err, remotePath, stream, resolve);
+          } else if (localPath.split(p.sep)[0] == "golang") {
+            // Run `go build` when a Golang file has changed
+            golangTask(err, remotePath, stream, resolve);
+          } else {
+            resolve();
+          }
+        });
+      });
+
+      writeStream.on("error", (err) => {
+        reject(err);
+      });
+
+      readStream.pipe(writeStream);
+    });
+  });
+}
+
+watcher.on("add", async (path, stats) => {
+  process.stdout.write(`${new Date().toLocaleString()} [+] ${path}`);
+  try {
+    await uploadFile(
+      path,
+      p.posix.join(config.deployDirectory, path.replace(p.sep, p.posix.sep))
+    );
+  } catch (err) {
+    console.error(" ✘", err);
+  }
+});
+
+watcher.on("change", async (path, stats) => {
+  process.stdout.write(`${new Date().toLocaleString()} [~] ${path}`);
+  try {
+    await uploadFile(
+      path,
+      p.posix.join(config.deployDirectory, path.replace(p.sep, p.posix.sep))
+    );
+  } catch (err) {
+    console.error(" ✘", err);
+  }
+});
+
+watcher.on("unlink", async (path, stats) => {
+  process.stdout.write(`${Date.now().toLocaleString()} [-] ${path}`);
+  try {
+    await deleteFile(
+      p.posix.join(config.deployDirectory, path.replace(p.sep, p.posix.sep))
+    );
+  } catch (err) {
+    console.error(" ✘", err);
+  }
+});
+
+console.log("watching for changes...");
+setInterval(() => {}, 1e6);
```

Comment:
Nitpick: Should we clear the screen at some point? 😋 
just wondering since the npm run watch tends to clear the screen on a few occasions 😋